src/liboslexec/llvm_instance.cpp

```diff
@@ -1324,12 +1327,37 @@ BackendLLVM::run ()
 #endif
 
 #ifdef OSL_LLVM_NO_BITCODE
+    OSL_ASSERT(!use_rs_bitcode());
     ll.module (ll.new_module ("llvm_ops"));
 #else
     if (! use_optix()) {
-        ll.module (ll.module_from_bitcode ((char*)osl_llvm_compiled_ops_block,
-                                           osl_llvm_compiled_ops_size,
-                                           "llvm_ops", &err));
+        if(use_rs_bitcode()){
+            ll.module (ll.module_from_bitcode ((char*)osl_llvm_compiled_rs_dependant_ops_block,
+                                               osl_llvm_compiled_rs_dependant_ops_size,
+                                               "llvm_rs_dependant_ops", &err));
+            if (err.length())
+                shadingcontext()->errorf("ParseBitcodeFile returned '%s'\n", err);
+
+
+            std::vector<char>& rs_free_function_bitcode = shadingsys().m_rs_bitcode;
+            OSL_ASSERT (rs_free_function_bitcode.size() && "Free Function bitcode is empty");
+            
+            llvm::Module* rs_free_functions_module =
+            ll.module_from_bitcode (static_cast<const char*>(rs_free_function_bitcode.data()),
+                                    rs_free_function_bitcode.size(), "rs_free_functions");
+            if (err.length())
```

Comment:
You are checking for an error message, but not passing `err` to the preceding `module_from_bitcode` call.

Reply:
@steenax86 we could add that check and error report to all module_from_bitcode calls (there are 4 others here) as a general improvement.

Reply:
I think it's up to you, @steenax86, whether to fix that here or just incorporate into a subsequent set of changes.

Reply:
@lgritz we added the error checks.